### PR TITLE
ref(sdk): Bump SDK to `2.0.0rc4` for dogfooding

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
This change also bumps `pytest-sentry` to `0.2.0`, which is compatible with versions 2.0+ of the SDK.

This PR is the successor of #66504, which was reverted in https://github.com/getsentry/sentry/commit/e477575ed35a523060419084300acf4fb819bea9. This PR addresses the [concerns](https://github.com/getsentry/sentry/pull/66504#discussion_r1551637212) leading to our decision to revert #66504.

### CODE REVIEW

**Comment by colin-sentry** [comment]
There are still hubs in a few places like relay/config/__init__.py:

```
from sentry_sdk import Hub, capture_exception
```

Are those going to be removed in a later commit?

**Comment by szokeasaurusrex** [comment]
I don't think we were planning to remove them yet @colin-sentry. The old hub API is deprecated, but it still works fine. Internally, the Hub API calls the equivalent new Scope API

File: requirements-base.txt [hunk 1]
```diff
@@ -67,7 +67,7 @@ sentry-kafka-schemas>=0.1.65
 sentry-ophio==0.2.6
 sentry-redis-tools>=0.1.7
 sentry-relay>=0.8.56
-sentry-sdk==2.0.0rc4
+sentry-sdk>=2.0.0rc4,<3
```
[the remark this answers is not in the document]
Comment (szokeasaurusrex) [reply requirements-base.txt]: fixed

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**